cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sinks/TableSink.java

```diff
@@ -16,30 +16,48 @@
 
 package co.cask.cdap.templates.etl.batch.sinks;
 
-import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.dataset.lib.KeyValue;
 import co.cask.cdap.api.dataset.table.Put;
 import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.templates.etl.api.Emitter;
 import co.cask.cdap.templates.etl.api.Property;
 import co.cask.cdap.templates.etl.api.StageConfigurer;
 import co.cask.cdap.templates.etl.api.batch.BatchSinkWriter;
 import co.cask.cdap.templates.etl.api.config.ETLStage;
+import co.cask.cdap.templates.etl.common.RecordPutTransformer;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
 
 /**
  * CDAP Table Dataset Batch Sink.
  */
-public class TableSink extends BatchWritableSink<Put, byte[], Put> {
+public class TableSink extends BatchWritableSink<StructuredRecord, byte[], Put> {
+  private RecordPutTransformer recordPutTransformer;
 
   @Override
   public void configure(StageConfigurer configurer) {
     configurer.setName("TableSink");
     configurer.setDescription("CDAP Table Dataset Batch Sink");
-    configurer.addProperty(new Property(NAME, "Name of the table. If the table does not already exist," +
-      " one will be created.", true));
-    configurer.addProperty(
-      new Property(DatasetProperties.SCHEMA,
-                   "Optional schema of the table as a JSON Object. If the table does not already exist," +
-                     " one will be created with this schema, which will allow the table to be explored through Hive.",
-                   false));
+    configurer.addProperty(new Property(
+      NAME, "Name of the table. If the table does not already exist, one will be created.", true));
+    configurer.addProperty(new Property(
+      Table.PROPERTY_SCHEMA,
+      "Optional schema of the table as a JSON Object. If the table does not already exist," +
+        " one will be created with this schema, which will allow the table to be explored through Hive.",
+      false));
+    configurer.addProperty(new Property(
+      Table.PROPERTY_SCHEMA_ROW_FIELD,
+      "the name of the record field that should be used as the row key when writing to the table.",
+      true));
+  }
+
+  @Override
+  public void initialize(ETLStage stageConfig) {
```

Comment:
Should we allow it to throw exception in initialize method similar to initialize method in RealtimeSource/RealtimeSink and CDAP programs?